core/opl/shovel.py

```diff
@@ -83,18 +120,178 @@ def args(parser, group_actions):
+                    ]
+                    if test_run_matcher == "null":
+                        print(
+                            f"WARNING: Test run {test_run_id} for test {args.test_name_horreum}/{test_id} does not have {args.test_job_matcher}, skipping"
+                        )
+                        continue
+                    if test_matcher == test_run_matcher:
+                        raise Exception(
+                            "INFO: Test result found in Horreum, skipping upload"
+                        )
+
+            print("INFO: Uploading to Horreum ... ")
+            params = {
+                "test": args.test_name_horreum,
+                "start": test_start,
+                "stop": test_end,
+                "owner": args.test_owner,
+                "access": args.test_access,
+            }
+            requests.post(
+                f"{args.horreum_host}/api/run/data",
+                params=params,
+                headers=headers,
+                data=json.dumps(values),
+            )
 
-    def result(self):
+    def result(self, args):
         print("Hello from pluginHorreum result")
+        if args.id_array is None:
+            raise Exception(
+                "Id array json file is needed to work with --horreum-result"
+            )
+        elif args.horreum_data_file is None:
+            raise Exception(
+                "Horreum data file is required to work with --horreum-result"
+            )
+        else:
+            values = json.loads(args.id_array)
+            is_fail = 0
+            for i in values:
+                id_value = i["id"]
+                jsonFile = open(args.horreum_data_file, "r")
+                values = json.load(jsonFile)
+                jsonFile.close()
+                test_start = values["timestamp"]
+                if test_start == "None":
+                    test_start = values["startTimestamp"]
+                test_end = values["endTimestamp"]
+                range_data = {
+                    "range": {
+                        "from": test_start,
+                        "to": test_end,
+                        "oneBeforeAndAfter": True,
+                    }
+                }
+
+                # Create a dictionary with the annotation query
+                annotation_data = {"annotation": {"query": id_value}}
+
+                # Write the range data to a JSON file
+                with open("/tmp/annotationQuery.json", "w") as file:
+                    json.dump(range_data, file)
+
+                # Append the annotation data to the JSON file
+                with open("/tmp/annotationQuery.json", "a") as file:
+                    json.dump(annotation_data, file)
+
+                # Send a POST request to the API and retrieve the result using curl and jq
+                curl_command = f"curl https://{args.horreum_host}/api/changes/annotations -s -H 'content-type: application/json' -d @/tmp/annotationQuery.json | jq -r ."
```

Comment:
Use Python `requests` library here. With that, you do not need to create a file.